R/delineate.R

```diff
@@ -37,20 +40,19 @@ delineate_corridor <- function(
     initial_method, capping_method
   )
 
-  if (segments) delineate_segments()
+  if (segments) {
+    # Select the relevant part of the network
+    buffer_corridor <- 100  # TODO should this be an additional input parameter?
+    corridor_buffer <- sf::st_buffer(corridor, buffer_corridor)
+    network_filtered <- filter_network(network, corridor_buffer)
+
+    corridor <- segments(corridor, network_filtered, osm_data$river_centerline,
+                         angle_threshold)
+  }
   if (riverspace) delineate_riverspace()
   return(corridor)
 }
 
-
```

Comment:
I thought this interface, with a single access point to both the corridor and segmentation would fit better, since one actually needs the same input for both delineation (the riverspace also requires the buildings). What do you think @cforgaci ?